src/hypertable.c

```diff
@@ -3014,3 +3020,125 @@ ts_hypertable_update_dimension_partitions(const Hypertable *ht)
 
 	return false;
 }
+
+/*
+ * hypertable_osm_range_update
+ * 0 hypertable REGCLASS,
+ * 1 range_start,
+ * 2 range_end,
+ * 3 empty=false
+ * If empty is set to true then the range will be set to invalid range
+ * but the overlap flag will be unset, indicating that no data is managed
+ * by OSM and therefore timescaledb optimizations can be applied.
+ */
+TS_FUNCTION_INFO_V1(ts_hypertable_osm_range_update);
+Datum
+ts_hypertable_osm_range_update(PG_FUNCTION_ARGS)
+{
+	Oid relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Hypertable *ht;
+	const Dimension *time_dim;
+	Cache *hcache;
+
+	Oid time_type; /* required for resolving the argument types, should match the hypertable
+					  partitioning column type */
+
+	hcache = ts_hypertable_cache_pin();
+	ht = ts_resolve_hypertable_from_table_or_cagg(hcache, relid, true);
+	Assert(ht != NULL);
+	time_dim = hyperspace_get_open_dimension(ht->space, 0);
+
+	if (time_dim == NULL)
+		elog(ERROR,
+			 "could not find time dimension for hypertable %s.%s",
+			 quote_identifier(NameStr(ht->fd.schema_name)),
+			 quote_identifier(NameStr(ht->fd.table_name)));
+
+	time_type = ts_dimension_get_partition_type(time_dim);
+
+	int32 osm_chunk_id = ts_chunk_get_osm_chunk_id(ht->fd.id);
+	if (osm_chunk_id == INVALID_CHUNK_ID)
+		elog(ERROR,
+			 "no OSM chunk found for hypertable %s.%s",
+			 quote_identifier(NameStr(ht->fd.schema_name)),
+			 quote_identifier(NameStr(ht->fd.table_name)));
+
+	int32 dimension_slice_id = ts_chunk_get_osm_slice_id(osm_chunk_id, time_dim->fd.id);
+
+	/*
+	 * range_start, range_end arguments must be converted to internal representation
+	 * a NULL start value is interpreted as INT64_MAX - 1 and a NULL end value is
+	 * interpreted as infinity.
+	 * Passing both start and end NULL values will reset the range to the default range an
+	 * OSM chunk is given upon creation, which is [INT64_MAX - 1, INT64_MAX]
+	 */
+	int64 range_start_internal, range_end_internal;
```

Comment:
i think we should require for range_start and range_end to be both NULL or none NULL and error otherwise